### PR TITLE
PIM-7373 Avoid deleting and reinsert all attributes relation at family save

### PR DESCRIPTION
**Pull request by BitOne**
This PR avoid generating lots of DELETE and INSERT queries by removing or adding only attributes that have been removed or added to the family at save time.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Done
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Done
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 1]
```diff
@@ -336,12 +336,29 @@ protected function createAttributeRequirement(
      */
     protected function addAttributes(FamilyInterface $family, array $data)
     {
+        $currentAttributeCodes = [];
+        $wantedAttributeCodes = [];
+
+        foreach ($data as $attributeCode) {
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: is it necessary? can't we just name the parameter "data" as "wantedAttributeCodes" directly?

File: CHANGELOG-1.7.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# 1.7.x
+
+## Bug fixes
+ - SDS-3606: Fix deletion and reinsertion of all attributes relations at family save time
```
Comment (ahocquard) [inline CHANGELOG-1.7.md]: PIM-7373

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php (hunk 1 above)
Comment (jmleroux) [reply src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: or `$wantedAttributeCodes = array_values($data);` if it's just a matter of key.

**Comment by LaurentPetard** [comment]
There's a side-effect to the fix : there's no versioning on the family if only attribute requirements are changed. 

In this case the Doctrine unit of work doesn't detect any change on the family. So I added a "Guesser" to mark the family as updated when an attribute requirement is added or removed.

**Pull request merged**